src/Mod/Start/Gui/FileCardDelegate.cpp

```diff
@@ -81,6 +109,40 @@ void FileCardDelegate::paint(QPainter* painter,
     }
     thumbnail->setFixedSize(thumbnailSize, thumbnailSize);
     thumbnail->setSizePolicy(QSizePolicy::Policy::Fixed, QSizePolicy::Policy::Fixed);
+
+    if (option.state & QStyle::State_Selected) {
+        QColor color = getSelectionColor();
+        widget->setStyleSheet(QString::fromLatin1("QWidget#thumbnailWidget {"
+                                                  " border: 2px solid rgb(%1, %2, %3);"
+                                                  " border-radius: 4px;"
+                                                  " padding: 2px;"
+                                                  "}")
+                                  .arg(color.red())
+                                  .arg(color.green())
+                                  .arg(color.blue()));
+    }
+    else if (option.state & QStyle::State_MouseOver) {
+        QColor color = getBorderColor();
+        widget->setStyleSheet(QString::fromLatin1("QWidget#thumbnailWidget {"
+                                                  " border: 2px solid rgb(%1, %2, %3);"
+                                                  " border-radius: 4px;"
+                                                  " padding: 2px;"
+                                                  "}")
+                                  .arg(color.red())
+                                  .arg(color.green())
+                                  .arg(color.blue()));
+    }
+    else if (qApp->styleSheet().isEmpty()) {

```

Comment:
In MSVC this needs `#include <QApplication>` added to PreCompiled.h, otherwise `qApp` resolves to a `QGuiApplication` instead of `QApplication`.